### ISSUE TITLE
test.each no-done-callback false positive in version 24.1.2

### ISSUE THREAD

**Issue by voda** [jest-community/eslint-plugin-jest#710]
Very similar to #709, but there is still a false positive when the test function has 2 arguments.

Plugin version: 24.1.2
Last working version: 24.1.0
Rule jest/no-done-callback has false positives with test.each

```ts
it.each([
	[1, 2],
	[2, 4],
	[3, 6],
])('.double(%i)', async (a: number, expected: number) => {
	expect(2 * a).toBe(expected);
});
```

**Comment by jsaguet** [comment]
I just had the same issue.

After having a quick look at the rule implementation, it seems the logic is to check the number of arguments of the test function.

But for `test.each` / `it.each`, just checking the number or arguments of the function will never work.
I think the only way to enforce this rule on `test.each` is to check if the last argument is actually called in the function block statement.

**Comment by voda** [comment]
> I think the only way to enforce this rule in this case is to check if the last argument is actually called in the function block statement.

The list of test cases can also contain a callback, e.g.:
```js
it.each([
	[1, () => 2],
])(...);
```

I think jest is also checking number of arguments: https://github.com/facebook/jest/blob/f9132f9106e3f1b3ae6f81d1f36222055c3fe111/packages/jest-each/src/bind.ts#L81-L83
